lib/metasploit/framework/credential_collection.rb

```diff
@@ -219,6 +226,9 @@ def each_unfiltered
       prepended_creds.each { |c| yield c }
 
       if username.present?
+        if usernames_only
```

Comment:
In this current iteration we'll only perform pre-authentication without any passwords being involved, but in a future iteration we'll also want to try passwords if the username exists.

Would this suggested pattern cater to those future requirements? 👀

Reply:
Consider instead of adding a new attribute a `filter` block passed in could be used to only provide unique user names.  It may be slightly more costly during runtime but would avoid introducing modes to this class that change the behaviour of `each_unfiltered`.

Reply:
Just syncing up with @sjanusz-r7 on this, and I think what we want is closer to `nil_passwords` here - similar to `blank_passwords` 🤔